janus_server/src/config.rs

```diff
@@ -119,81 +108,60 @@ pub struct AggregationJobCreatorConfig {
 }
 
 impl BinaryConfig for AggregationJobCreatorConfig {
-    fn database_config(&mut self) -> &mut DbConfig {
-        &mut self.database
-    }
-
-    fn logging_config(&mut self) -> &mut TraceConfiguration {
-        &mut self.logging_config
-    }
-
-    fn metrics_config(&mut self) -> &mut MetricsConfiguration {
-        &mut self.metrics_config
+    fn common_config(&mut self) -> &mut CommonConfig {
+        &mut self.common_config
     }
 }
 
-/// Non-secret configuration options for the Janus Aggregation Job Driver job.
+/// Non-secret configuration options for Janus Job Driver jobs.
 ///
 /// # Examples
 ///
 /// ```
-/// use janus_server::config::AggregationJobDriverConfig;
+/// use janus_server::config::JobDriverConfig;
 ///
 /// let yaml_config = r#"
 /// ---
 /// database:
 ///   url: "postgres://postgres:postgres@localhost:5432/postgres"
 /// logging_config: # logging_config is optional
 ///   force_json_output: true
-/// min_aggregation_job_discovery_delay_secs: 10
-/// max_aggregation_job_discovery_delay_secs: 60
-/// max_concurrent_aggregation_job_workers: 10
-/// aggregation_worker_lease_duration_secs: 600
-/// aggregation_worker_lease_clock_skew_allowance_secs: 60
+/// min_job_discovery_delay_secs: 10
+/// max_job_discovery_delay_secs: 60
+/// max_concurrent_job_workers: 10
+/// worker_lease_duration_secs: 600
+/// worker_lease_clock_skew_allowance_secs: 60
 /// "#;
 ///
-/// let _decoded: AggregationJobDriverConfig = serde_yaml::from_str(yaml_config).unwrap();
+/// let _decoded: JobDriverConfig = serde_yaml::from_str(yaml_config).unwrap();
 /// ```
 #[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
-pub struct AggregationJobDriverConfig {
-    /// Configuration for the database backend to connect to.
-    pub database: DbConfig,
-    /// Logging configuration.
-    #[serde(default)]
-    pub logging_config: TraceConfiguration,
-    #[serde(default)]
-    pub metrics_config: MetricsConfiguration,
-    /// The minimum delay between checking for aggregation jobs ready to be stepped, in seconds.
-    /// Applies only when there are no aggregation jobs to be stepped.
-    pub min_aggregation_job_discovery_delay_secs: u64,
-    /// The maximum delay between checking for aggregation jobs ready to be stepped, in seconds.
-    /// Applies only when there are no aggregation jobs to be stepped.
-    pub max_aggregation_job_discovery_delay_secs: u64,
-    /// The maximum number of aggregation jobs being stepped at once. This parameter determines the
-    /// amount of per-process concurrency.
-    pub max_concurrent_aggregation_job_workers: usize,
-    /// The length of time, in seconds, workers will acquire a lease for the aggregation jobs they
-    /// are stepping. Along with aggregation_worker_lease_clock_skew_allowance, determines the
-    /// effective timeout of stepping a single aggregation job.
-    pub aggregation_worker_lease_duration_secs: u64,
+pub struct JobDriverConfig {
```

Comment:
I'm not sure we want to make `aggregation_job_driver` & `collect_job_driver` share the same config files, in case they need different options in the future.

OTOH, the "driver" options could be usefully factored out to their own sub-structure, included (perhaps with `#[serde(flatten)]`) in both job-specific configurations.